flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyedStateBackend.java

```diff
@@ -61,8 +61,9 @@
 	 * 		   over it keys are not supported.
 	 * @param state State variable for which existing keys will be returned.
 	 * @param namespace Namespace for which existing keys will be returned.
+	 * @param namespaceSerializer the serializer for the namespace.
 	 */
-	<N> Stream<K> getKeys(String state, N namespace);
+	<N> Stream<K> getKeys(String state, N namespace, TypeSerializer<N> namespaceSerializer);
```

Comment:
I noticed that introducing this additional parameter is actually not required. It is only used in RocksDB, where we can also get the namespace serializer in `getKeys(...)` via
```
		Tuple2<ColumnFamilyHandle, RegisteredKeyedBackendStateMetaInfo<?, ?>> columnInfo = kvStateInformation.get(state);
		if (columnInfo == null) {
			return Stream.empty();
		}
...
		columnInfo.f1.getNamespaceSerializer();
```

Reply:
Aha, I also noticed that `RegisteredKeyedBackendStateMetaInfo` contains the namespace serializer. But there is one thing that confuses me, I found that it looks like one `ColumnFamilyHandle` can correspond to multiple namespace serializer but I am not sure about this, so I introduced this additional parameter. Am I misunderstanding something? Can one `ColumnFamilyHandle` only correspond to one namespace serializer?

Reply:
Currently, each registered state has its own column family, so that should be ok. What made you think that this does not hold? (asking in case you found something that should be fixed)